### PR TITLE
bazel - add, cleanup baseline metrics

### PR DESCRIPTION
**Pull request by dgdonut**
Bazel API endpoint metrics split into success/failure counts. Record histograms of per-request data (read bytes, write bytes, task times, etc).

### CODE REVIEW

File: bazel/cas/service.go [hunk 1]
```diff
@@ -154,15 +181,17 @@ func (s *casServer) Read(req *bytestream.ReadRequest, ser bytestream.ByteStream_
 		log.Infof("Opening store resource for reading: %s", storeName)
 		r, err = s.storeConfig.Store.OpenForRead(storeName)
 		if err != nil {
-			log.Errorf("Failed to OpenForRead: %v", err)
-			return status.Error(codes.NotFound, fmt.Sprintf("Failed opening resource %s for read, returning NotFound. Err: %v", storeName, err))
+			// Reset err to nil - don't count NotFound towards request failures
+			openErr := err
```
Comment (ryancouto) [inline bazel/cas/service.go]: this reads a little weird... not sure how to clean it up though. maybe just a little more context in the comment as to why this isn't counted as a request failure.

File: bazel/cas/service.go [hunk 2]
```diff
@@ -197,6 +226,7 @@ func (s *casServer) Read(req *bytestream.ReadRequest, ser bytestream.ByteStream_
 		if err == nil {
 			continue
 		} else if err == io.EOF {
+			err = nil
```
Comment (ryancouto) [inline bazel/cas/service.go]: why?

File: bazel/cas/service.go [hunk 3]
```diff
@@ -403,8 +454,11 @@ func (s *casServer) GetActionResult(ctx context.Context,
 
 	r, err := s.storeConfig.Store.OpenForRead(address.storeName)
 	if err != nil {
-		log.Errorf("Failed to OpenForRead: %v", err)
-		return nil, status.Error(codes.NotFound, fmt.Sprintf("Failed opening %s for read, returning NotFound. Err: %v", address.storeName, err))
+		// Reset err to nil - don't count NotFound towards request failures
```
Comment (ryancouto) [inline bazel/cas/service.go]: ditto

**Review by ryancouto — APPROVED** [review]
Looks good, could use a couple more explanatory comments but 👍

**Comment by dgdonut** [comment]
Updated comments around instances where we reset err to nil

**Pull request merged**